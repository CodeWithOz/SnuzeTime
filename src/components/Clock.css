@font-face {
  font-family: 'Digital7';
  src: url('../fonts/Digital-7MonoItalic.eot?#iefix')
      format('embedded-opentype'),
    url('../fonts/Digital-7MonoItalic.svg#Digital-7MonoItalic') format('svg'),
    url('../fonts/Digital-7MonoItalic.ttf') format('truetype'),
    url('../fonts/Digital-7MonoItalic.woff') format('woff'),
    url('../fonts/Digital-7MonoItalic.woff2') format('woff2');
}

.clock-container .border {
  border: 3px solid red;
  border-radius: 10px;
  display: inline-block;
  padding: 0.2em;
}

.clock-container .time {
  /* Custom font not used due to FOUT/FOIT issues */
  /* Will use after I have satisfactorily dealt with the issues */
  font-family: Arial, sans-serif;
}
