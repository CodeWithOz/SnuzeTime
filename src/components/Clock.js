import './Clock.css';
import React from 'react';

const Clock = props => {
  const [, time, meridiem] = props.currentTime.match(
    /(\d{2}:\d{2}:\d{2}) (AM|PM)/
  );

  return (
    <div className="clock-container">
      <div className="border">
        <span className="time">{time}</span> {meridiem}
      </div>
    </div>
  );
};

export default Clock;
