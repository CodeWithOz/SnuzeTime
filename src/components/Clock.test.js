import React from 'react';
import Clock from './Clock';
import renderer from 'react-test-renderer';

test('Clock component renders correctly', () => {
  const clock = renderer.create(<Clock currentTime={'06:14 AM'} />);

  // render the component
  const tree = clock.toJSON();
  expect(tree).toMatchSnapshot();
});
