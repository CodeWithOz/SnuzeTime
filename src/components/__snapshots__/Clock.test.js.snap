// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock component renders correctly 1`] = `
<div>
  06:14 AM
</div>
`;
